src/NextGen/DonationForm/Blocks/DonationFormBlock/resources/editor/Edit.tsx

```diff
@@ -13,10 +13,16 @@ import {useCallback} from 'react';
 /**
  * @since 1.0.0
  */
-export default function Edit({attributes, setAttributes}: BlockEditProps<any>) {
-    const {formId} = attributes;
+export default function Edit({clientId, attributes, setAttributes}: BlockEditProps<any>) {
+    const {formId, blockId} = attributes;
     const {formOptions, isResolving} = useFormOptions();
 
+    useEffect(() => {
+        if (!blockId) {
+            setAttributes({blockId: clientId});
```

Comment:
Have we looked into the possibility of conflicts with re-using the `clientId` as a unique identifier outside of the block editor?

The `clientId` is used to track the block within the editor and is not persisted in the post content. This value changes each time the post editor is loaded, so the uniqueness of the value is limited to editor session.

Reply:
I'm only using it as an additional data attribute on the iframe being rendered to target our form for displaying a receipt.  They only have to be unique for our blocks specifically, so i'm not worried about conflicts.